src/Java.Interop.Tools.JavaTypeSystem/Adapters/ManagedApiImporter.cs

```diff
@@ -0,0 +1,379 @@
+
+			return model;
+		}
+
+		static bool ShouldImport (TypeDefinition td)
+		{
+			// We want to exclude "IBlahInvoker" and "IBlahImplementor" and "BlahConsts" types
+			if (td.Name.EndsWith ("Invoker")) {
+				var n = td.FullName;
+				n = n.Substring (0, n.Length - 7);
+
+				var types = td.DeclaringType != null ? td.DeclaringType.Resolve ().NestedTypes : td.Module.Types;
+
+				if (types.Any (t => t.FullName == n))
+					return false;
+			}
+
+			if (td.Name.EndsWith ("Implementor")) {
+				var n = td.FullName;
+				n = n.Substring (0, n.Length - 11);
+
+				var types = td.DeclaringType != null ? td.DeclaringType.Resolve ().NestedTypes : td.Module.Types;
+
+				if (types.Any (t => t.FullName == n))
+					return false;
+			}
+
+			if (td.Name.EndsWith ("Consts")) {
+				var n = td.FullName;
+				n = n.Substring (0, n.Length - 6);
+
+				var types = td.DeclaringType != null ? td.DeclaringType.Resolve ().NestedTypes : td.Module.Types;
+
+				if (types.Any (t => t.FullName == n))
+					return false;
+			}
+
+			return true;
+		}
+
+		public static JavaClassModel? ParseClass (TypeDefinition type, JavaTypeCollection collection)
+		{
+			// TODO: type parameters?
+			var obs_attr = GetObsoleteAttribute (type.CustomAttributes);
+			var reg_attr = GetRegisterAttribute (type.CustomAttributes);
+
+			if (reg_attr is null)
+				return null;
+
+			var encoded_fullname = ((string) reg_attr.ConstructorArguments [0].Value).Replace ('/', '.');
+			var (package, nested_name) = DecodeRegisterJavaFullName (encoded_fullname);
+
+			var base_jni = GetBaseTypeJni (type);
+
+			var model = new JavaClassModel (
+				javaPackage: GetOrCreatePackage (collection, package, type.Namespace),
+				javaNestedName: nested_name,
+				javaVisibility: type.IsPublic || type.IsNestedPublic ? "public" : "protected internal",
+				javaAbstract: type.IsAbstract,
+				javaFinal: type.IsSealed,
+				javaBaseType: base_jni.Replace ('/', '.').Replace ('$', '.'),
+				javaBaseTypeGeneric: base_jni.Replace ('/', '.').Replace ('$', '.'),
+				javaDeprecated: obs_attr != null ? "deprecated" : "not-deprecated",
+				javaStatic: false,
+				jniSignature: FormatJniSignature (package, nested_name),
+				baseTypeJni: base_jni.HasValue () ? $"L{base_jni};" : string.Empty
+			); ;
+
+			ParseImplementedInterfaces (type, model);
+
+			foreach (var method in type.Methods.Where (m => !m.IsConstructor))
+				if (ParseMethod (method, model) is JavaMethodModel m)
+					model.Methods.Add (m);
+
+			return model;
+		}
+
+		public static JavaInterfaceModel? ParseInterface (TypeDefinition type, JavaTypeCollection collection)
+		{
+			// TODO: type paramters?
+			var obs_attr = GetObsoleteAttribute (type.CustomAttributes);
+			var reg_attr = GetRegisterAttribute (type.CustomAttributes);
+
+			if (reg_attr is null)
+				return null;
+
+			var encoded_fullname = ((string) reg_attr.ConstructorArguments [0].Value).Replace ('/', '.');
+			var (package, nested_name) = DecodeRegisterJavaFullName (encoded_fullname);
```

Comment:
Not sure why you want to give `DecodeRegisterJavaFullName()` the `encoded_fullname` value which replaces `/` with `.`, instead of having `DecodeRegisterJavaFullName()` deal with the `/`'s on its own.